### ISSUE TITLE
stuck at step zero

### ISSUE THREAD

**Issue by alto777** [inkstitch/inkstitch#263]
I've used Inkscape and a friend got an embroidery moochine, so I'm just test driving Inkstitch.

I was setting it up to run in Windows (iMac w/ Parallels Desktop), when I went for the Inkstitch I was delighted to find a Macintosh version since last time I looked.

I tried placing the unzipped flower in the extensions area, no joy. Placing all the files in the unzipped folder into the extension area gives me "Embroidery" in my Extensions menu.

When I try Extensions/Embroidery/Params, I get an error dialog box

Inkscape has received additional data from the script executed.  The script did not return an error, but this may indicate the results will not be as expected.

containing

Traceback (most recent call last):
  File "inkstitch.py", line 33, in <module>
    extension = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

I am using Inkscape  0.92.2 5c3e80d, 2017-08-06

Thinking it might be an issue with the newish Mac OS version of Inkstitch, I completed the same steps under Windows, that's new territory for me and Inkscape so the Inkscape version is the latest downloaded this day.

The Windows setup fails in the same way.

I scanned around here and will continue to do. I think I am missing something obvious. Feel free to slap with with a large cold trout while also suggesting something I might try or 'splaining what I've stepped over in my excitement and impatience.

alto777

**Comment by lexelby** [comment]
This is a no trout zone! :) Unless you want to embroider one, of course.

It sounds like there's something not quite right in the unpacking.  I'd recommend you remove what you put in the extensions directory and start over.

You should end up with something like this:

```
/path/to/inkscape/extensions/
                             inkstitch.py
                             inkstitch_embroider.inx
                             <lots more inx files>
                             inkstitch/
                                       bin/
                                           inkstitch
                                           <tons of code library files and folders>
                                       examples/
                                           <svg files>
                                       palettes/
                                           <.gpl files>
                                       symbols/
                                           inkstitch.svg
```

The key is that the executable and library files need to be under `inkstitch/bin/` inside your extensions folder.

Does that help?  If not, perhaps we can produce an installation video or something.

**Comment by lexelby** [comment]
Well, that sure is a puzzle!  It looks like you did exactly right.

Do you by any chance know any Python?  The next thing I'd want to try would be to modify inkstitch.py to tell us what file it's trying to execute, to make sure it's getting the path right. If you don't know any Python, no problem, I'll post a new version of the file here.

**Comment by alto777** [comment]
tl;dr - I have passed the stumbling block! At least on Windows.

I thought also of looking at the Python to strengthen the error reporting - not a Python programmer but I can read most code and have had some luck with things like this.

We have several paths (!) to the goal that might be simpler. On the other hand, it rankles that what works for most is not (yet) working for me.

If it comes to it, I will take a try at the Python and then come back for help.

Update: File under clutching at straws.

I tried again on the Mac OS side using the command line to move Inkstich components into the extensions folder thinking it might be a drag/drop copy problem. No joy.

On the Windows side, I unzipped the inkstitch-v1.15.0-win32.zip by double-clicking in Windows. I am no Windows expert but it did open up the archive. I copied the files/directories to the extensions folders and

I am now on the air with Inkstitch! Now have to learn how to use it, but I did see the parameters window and the preview played out on a little Bezier doodle I made for it to test.

I do not know why this would make a difference; perhaps it is not what (or all) that it took. I would much prefer to get the Mac OS version working, and I will think of a way to repeat what I tried without literally doing exactly the same thing.

But any port in a storm. Inkscape is Inkscape, *.svg files are platform independent, so we can arrange a workflow involving the Windows version and get on w/ the embroidery.

I am prepared but hope another odd error doesn't crop. For now it is just getting familiar with Inkstitch and learning about the subtleties of creating a stitch file from vector art.

My friend did buy a few designs, that is she had some designs "digitized". Naturally she wants to eliminate the time and expense and inflexibility of this approach.

I'll report any progress I make if I do work on the Mac OS side of things. Right now my friend does not have any way to run Windows but I will not mind seeing a bit more of her than I might otherwise.

alto777

**Comment by lexelby** [comment]
Glad to help.  Have fun!

I'm gonna reopen this so I don't forget to do the new release.

**Comment by alto777** [comment]
new tl;dr. I am on the air with Mac OS after (insanity) I think doing the same thing again…

1. I tired using Windows to unzip the Mac OS Inkstitch, but it just passed the buck so to speak back to Mac OS.

2. I tried WinZip but was warned about incompatibility and it seemed I had the files but not the structure and (?) many copies of an *.mo file it complained about. Gave up on WInZip.

3. I used the Mac CLI to do the unarchiving

 gunzip -c the.tar.gz | tar xopf -

it got all the files and (sorry, bad note taker) either this manner of getting Inkstitch where it needs to be or just plain double-clicking in desperation resulted in success, that is I see the params dialog, I have gone no further just now.

I will be able to do a total pristine Inkscape/Inkstitch installation on my friend's iMac. Right now it looks like something odd, _possibly_ related to the exact manner of decompressing, although I would expect a double-click to be doing no more or less than the CLI commands.

Also I was thrilled to see a *.pes file in Inkscape and it was nice to be able to re-kern the one flaw with the purchased digitized file!

I'll let you know that it worked (it will, it will) on the fresh iMac.

a7

**Comment by lexelby** [comment]
Thanks!  Looking forward to your next report.
